src/shared/telemetry/defaultTelemetryPublisher.ts

```diff
@@ -65,20 +67,27 @@ export class DefaultTelemetryPublisher implements TelemetryPublisher {
         this.telemetryClient = await DefaultTelemetryClient.createDefaultClient(
             this.clientId,
             this.region,
-            this.credentials
+            this.credentials,
+            this.trimmedAwsContext
         )
     }
 
-    public static async fromDefaultIdentityPool(clientId: string): Promise<IdentityPublisherTuple> {
-        return this.fromIdentityPool(clientId, DefaultTelemetryClient.DEFAULT_IDENTITY_POOL)
+    public static async fromDefaultIdentityPool(
```

Comment:
In general we should avoid static methods, and export module-level methods instead.

Reply:
Out of scope for this PR